core/src/main/java/com/linecorp/armeria/client/limit/DefaultConcurrencyLimit.java

```diff
@@ -65,7 +75,17 @@ int acquiredPermits() {
 
     @VisibleForTesting
     int availablePermits() {
-        return maxConcurrency - acquiredPermits.get();
+        int availablePermitCount = maxConcurrency() - acquiredPermits.get();
+        return Math.max(availablePermitCount, 0);
+    }
+
+    @VisibleForTesting
+    int maxConcurrency() {
+        int maxConcurrency = this.maxConcurrency.get();
+        if (maxConcurrency <= 0) {
+            return Integer.MAX_VALUE;
```

Comment:
What's the expected behavior when it produces a negative value?
Shouldn't we use just `0` instead of `MAX_VALUE`?
How about also logging with the debug level??